src/auslib/services/releases.py

```diff
@@ -297,8 +307,8 @@ def update_release(name, blob, old_data_versions, when, changed_by, trans):
             new_assets = current_assets[0]["data"]
```

Comment:
This is a good point actually. I poked at this a bit and I think we can improve readability and consistency with `set_release` by moving it outside of the loop.

The only downside is that we may be retrieving more data than needed if we're only updating a subset of assets. In practice, this is very unlikely though (the UI will always update all assets; balrogscript will always only update one).

Reply:
What code path will balrogscript end up using ?

Reply:
We'll be using PUTs to `/releases/{name}` for top level blob submission (but these requests won't contain any data that ends up in `release_assets`), and POSTs to `/releases/{name}` for locale submissions, which will end up writing one row to `release_assets`, and nothing to `releases_json`.

Reply:
OK, so my earlier comment was probably unhelpful, sorry. A platform-locale update via a balrogscript worker will look up a lot more with the `get_all_assets()` call than it would have done inside the loop. There's the UI case where it will probably speed things up, but that's a lot less common. 

Lots of ways we could move forward from here - monitor the db load in prod to see if it's actually a problem, go back to the original impl, determine what asset paths are changing in the submission and retrieve those only, etc.

Reply:
Let's see how this looks in practice before we take any further steps here.